### ISSUE TITLE
 Log rotation does not work

### ISSUE THREAD

**Issue by u4aew** [log4js-node/log4js-node#838]
log4js version: 4.0.2

Updated from version 3.0.6 to 4.0.2 log rotation stopped working
### Expected behavior
log rotation works

### Actual behavior
log rotation does not work

### How to reproduce
https://github.com/log4js-node/log4js-node/blob/v3.0.6/test/tap/dateFileAppender-test.js

it is enough to replace in the test configure with dateFileAppender
pattern on '-yyyy-mm-dd-hh-mm' and remove file deletion at the end
on

```
batch.test('configure with dateFileAppender', (t) => {
    log4js.configure({
      appenders: {
        date: {
          type: 'dateFile',
          filename: 'test/tap/date-file-test.log',
          pattern: '-yyyy-MM-dd-hh-mm',
          layout: { type: 'messagePassThrough' }
        }
      },
      categories: { default: { appenders: ['date'], level: 'WARN' } }
    });
    const logger = log4js.getLogger('tests');
    logger.info('this should not be written to the file');
    logger.warn('this should be written to the file');

    log4js.shutdown(() => {
      fs.readFile(path.join(__dirname, 'date-file-test.log'), 'utf8', (err, contents) => {
        t.include(contents, `this should be written to the file${EOL}`);
        t.equal(contents.indexOf('this should not be written to the file'), -1);
        t.end();
      });
    });

    // t.teardown(() => { removeFile('date-file-test.log'); });
  });
```

**Comment by nomiddlename** [comment]
The test you've got above will not rotate any files because you're only logging one message, so there is no time difference. Do you mean that when you run the test twice (without deleting old files), that you expect the previous file to be rotated rather than appended to?

**Comment by u4aew** [comment]
> The test you've got above will not rotate any files because you're only logging one message, so there is no time difference. Do you mean that when you run the test twice (without deleting old files), that you expect the previous file to be rotated rather than appended to?


The test must be run several times, the expected behavior is that in 2 minutes the rotation of the logs will occur, but it is not

**Comment by u4aew** [comment]
> The test you've got above will not rotate any files because you're only logging one message, so there is no time difference. Do you mean that when you run the test twice (without deleting old files), that you expect the previous file to be rotated rather than appended to?

**Comment by u4aew** [comment]
Any updates on this issue?

**Comment by nomiddlename** [comment]
I think this issue might be related to #842 where the existing file states are not taken into account when the app starts up.

**Comment by KonovalovaKV** [comment]
Hello Guys,
Did this issue still happen on the latest version?
It is very important for me

**Comment by nomiddlename** [comment]
It does still happen. There is a workaround - if you use the `alwaysIncludePattern: true` flag on your file appender config, then you won't see the issue.

**Comment by KonovalovaKV** [comment]
Any news?

**Comment by nomiddlename** [comment]
A fix was published to NPM in version 4.2.0 - please re-open this issue if the problem persists.

**Comment by KonovalovaKV** [comment]
It works for me, thanks!
